libs/libcommon/src/libcommon/storage_client.py

```diff
@@ -44,7 +46,7 @@ def __init__(
     ) -> None:
         logging.info(f"trying to initialize storage client with {protocol=} {storage_root=} {base_url=} {overwrite=}")
         self.storage_root = storage_root
-        self.protocol = protocol
+        self.protocol = cast(Literal["file", "s3"], protocol)
```

Comment:
maybe even set `protocol: Literal["file", "s3"]` in the constructor, and handle the parsing when we read the environment variables instead.

Reply:
The problem with that is that any service instantiating the `StorageClient` should have defined the passing `protocol` variable as `Literal` as well, but most of them define it as `str`. Therefore, mypy would complain:
```
error: Incompatible types in assignment (expression has type "str", variable has type "Literal['file', 's3']")  [assignment]
```

This way, any service can define `protocol` argument as `str` and we handle the cats within the `StorageClient` itself.

Reply:
I am refactoring the code...

Reply:
I have defined the StorageProtocol type and used it in AssetsConfig, CachedAssetsConfig and as parameter in StorageClient.

Reply:
Yes, I think it's better now. Extra thing (can be in another PR, even if it could enter here too): I think we should validate in the config. Maybe with `env.enum`: https://pypi.org/project/environs/, or the example they show below with marshmallow:

```
# using marshmallow validators
env.str(
    "NODE_ENV",
    validate=OneOf(
        ["production", "development"], error="NODE_ENV must be one of: {choices}"
    ),
)
```

wdyt?

Reply:
Done.

Reply:
Surely more nerd-sniping than a useful thing, but we could try to not repeat the list of valid strings, with something like https://stackoverflow.com/a/78297120/7351594 (only valid in 3.11 though). Alternative: declare the list of values along with `StorageProtocol` to be sure to update both if we support another protocol

```
StorageProtocol = Literal["file", "s3"]
STORAGE_PROTOCOL_VALUES: Tuple[StorageProtocol] = ["file", "s3"]
```

Feel free to ignore, though

Reply:
Done, although with a slightly different logic.